### PR TITLE
WIP Add permission view to admin inlines

### PR DESCRIPTION
**Pull request by szok**
It works:
display inline as readonly (all fields are readonly)

Not working:
Saving inlines. At the moment when you save a page from inline readonly django writes "None" for each fields.

### CODE REVIEW

File: src/ralph/lib/permissions/admin.py [hunk 1]
```diff
@@ -110,6 +110,19 @@ def get_list_display(self, request):
         ]
         return list_display
 
+    def get_inline_instances(self, request, obj=None):
+        inlines = super().get_inline_instances(request, obj)
+        new_ilines = []
```
Comment (ghost) [inline src/ralph/lib/permissions/admin.py]: typo ;)

File: src/ralph/lib/permissions/views.py [hunk 2]
```diff
@@ -55,6 +57,20 @@ def __new__(cls, name, bases, attrs):
...
 
+class PermissionInlineViewMetaClass(type):
+
+    """
+    Adding permission to additional inline views.
+    """
+
+    def __new__(cls, name, bases, attrs):
+        codename = 'can_view_inline_{}'.format(name.lower())
+        attrs['permision_codename'] = codename
```
Comment (mkurek) [inline src/ralph/lib/permissions/views.py]: permission

File: src/ralph/admin/mixins.py [hunk 3]
```diff
@@ -390,10 +393,30 @@ class RalphMPTTAdmin(MPTTModelAdmin, RalphAdmin):
...
+
+class RalphInlineMixin(object, metaclass=PermissionInlineMetaClass):
     # display change link for inline row in popup
     change_link_url_params = '_popup=1'
 
+    def get_formset(self, request, obj=None, **kwargs):
+        from django import forms
+        class ModelForm(forms.ModelForm):
+            def save(self, commit=True):
+                # import ipdb; ipdb.set_trace()
```
Comment (vi4m) [inline src/ralph/admin/mixins.py]: ?

**Comment by mkurek** [comment]
@szok could you describe status of this PR? what's currently working, what's not?

**Comment by szok** [comment]
@mkurek Done :)

**Comment by quamilek** [comment]
@szok What is status of this PR? Needs it some work? (build failed,`isort` showed errors)

**Comment by szok** [comment]
@quamilek It does not work properly with saved inlines. Saved data are empty.
We left doing it for future

**Comment by romcheg** [comment]
Closing this PR due to no activity for a long period of time.

**Pull request closed without merging**